component/componenttest/configtest.go

```diff
@@ -93,20 +97,17 @@ func checkStructFieldTags(f reflect.StructField) error {
 		}
 	}
 
-	// Check if squash is specified.
-	squash := false
 	for _, tag := range tagParts[1:] {
-		if tag == "squash" {
-			squash = true
-			break
-		}
-	}
-
-	if squash {
-		// Field was squashed.
-		if (f.Type.Kind() != reflect.Struct) && (f.Type.Kind() != reflect.Ptr || f.Type.Elem().Kind() != reflect.Struct) {
-			return fmt.Errorf(
-				"attempt to squash non-struct type on field %q", f.Name)
+		switch tag {
+		case "squash":
+			if (f.Type.Kind() != reflect.Struct) && (f.Type.Kind() != reflect.Ptr || f.Type.Elem().Kind() != reflect.Struct) {
+				return fmt.Errorf(
+					"attempt to squash non-struct type on field %q", f.Name)
+			}
+		case "remain":
+			if f.Type.Kind() != reflect.Map && f.Type.Kind() != reflect.Interface {
+				return fmt.Errorf(`attempt to use "remain" on non-map or interface type field %q`, f.Name)
+			}
```

Comment:
Do we have any uses of remain currently? Should we just forbid its usage entirely?

Reply:
We don't use it ourselves, but we do use it through the OTel SDK config package: https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/otelconf/v0.3.0/generated_config.go#L422.

In general I think it would make sense to advise against `remain` for Collector config structs, but I'm not sure there's a good way to do it. The best I can think of would be to update this function to take an option that enables support for it.